server/routes/offenceCodeDecisions/offenceCodeDecisionsValidation.ts

```diff
@@ -1,24 +1,74 @@
 import { FormError } from '../../@types/template'
+import { DecisionForm, SelectPrisonerData, SelectStaffData } from './decisionForm'
+import decisionTree from '../../offenceCodeDecisions/DecisionTree'
+import { DecisionType } from '../../offenceCodeDecisions/Decision'
 
-type DecisionForm = {
-  selectedDecisionId?: string
-}
-
-// eslint-disable-next-line no-shadow
-enum ErrorType {
-  MISSING_SELECTION = 'MISSING_SELECTION',
-}
-
-const errors: { [key in ErrorType]: FormError } = {
-  MISSING_SELECTION: {
-    href: '#selectedDecisionId',
-    text: 'Please make a choice', // TODO correct message
-  },
-}
-
-export default function validateForm({ selectedDecisionId }: DecisionForm): FormError | null {
+// What is valid depends on whether this is a normal submit or one searching for a user.
+// This method is tightly bound to the njk page validation names.
+export default function validateForm(decisionForm: DecisionForm, searching: boolean): FormError[] | null {
+  const { selectedDecisionId } = decisionForm
   if (!selectedDecisionId) {
-    return errors.MISSING_SELECTION
+    return [
+      {
+        href: '#selectedDecisionId',
+        text: 'Please make a choice',
+      },
+    ]
+  }
```

Comment:
Validation is all a bit complex as it depends on whether or not the user is searching or submitting, and whether they are choosing an option that requires a prisoner or a member of staff etc.